primo-zip.py

```diff
@@ -0,0 +1,77 @@
+# importing required modules
+import shutil
+import fileinput
+import json
+import os
+
+
+def swap_secrets_and_values(secret_data, types_to_check, directory, secret_visible=1):
+
+    # var for file paths list
+    file_paths = []
+
+    # crawl the directory structure
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            if (filename.endswith(types_to_check)):
+                filepath = os.path.join(root, filename)
+                file_paths.append(filepath)
+
+                # find and replace each secret tag with the actual secret
+                cnt = 0
+                for i in secret_data['secret_details']:
+                    secret_name = str(
+                        secret_data['secret_details'][cnt]['secret_name'])
+                    secret_value = str(
+                        secret_data['secret_details'][cnt]['secret_value'])
+
+                    secret_swap = [secret_name, secret_value]
+                    with fileinput.FileInput(filepath, inplace=True) as file:
+                        for line in file:
+                            print(line.replace(
+                                str(secret_swap[abs(1-secret_visible)]), secret_swap[secret_visible]), end='')
```

Comment:
@darrowcoucla This works - for `secret_visible` values of 0 and 1, at least - but could be clearer.  A comment before line 31 would help a lot: 
```
Replaces secret_name with secret_value when secret_visible = 1; replaces secret_value with secret_name when secret_visible = 0
```
Assuming I've understood it correctly, of course :)

If 0 and 1 are the only valid values for `secret_visible`, I think the `abs()` is not needed:  neither 1-0, nor 1-1, is ever negative.  If other values for `secret_visible` are allowed, under what circumstances?

Reply:
Comment added and abs() removed. It tests ok.

Reply:
pushing now.